core/src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -445,6 +445,10 @@ public void randomIndexTemplate() throws IOException {
                 mappings.endObject().endObject();
             }
 
+            for (String setting : randomSettingsBuilder.internalMap().keySet()) {
+                assertThat("non index. prefix setting set on index template, its a node setting...", setting, startsWith("index."));
```

Comment:
I really think the root cause of the problem here is that we accept this on the server maybe we can require all settings to start with `index.` on the server side? (different issue if you want)